src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -54,11 +54,18 @@
 import java.lang.invoke.MethodType;
 import java.util.Objects;
 
-public abstract sealed class AbstractLinker implements Linker permits LinuxAArch64Linker, MacOsAArch64Linker, AixPPC64Linker,
-                                                                      SysVPPC64leLinker, SysVS390xLinker, SysVx64Linker, WindowsAArch64Linker, Windowsx64Linker, LinuxRISCV64Linker {
+public abstract sealed class AbstractLinker implements Linker permits LinuxAArch64Linker, MacOsAArch64Linker,
+                                                                      AixPPC64Linker, SysVPPC64leLinker, SysVS390xLinker,
+                                                                      SysVx64Linker, WindowsAArch64Linker,
+                                                                      Windowsx64Linker, LinuxRISCV64Linker {
+
+    public interface UpcallStubFactory {
+        MemorySegment makeStub(MethodHandle target, SegmentScope arena);
+    }
```

Comment:
Can you please update this to merge with the latest openj9 content to ensure that this class is merged properly?
(My attempt to do so seemed to merge cleanly, but resulted in two definitions of this interface.)

Reply:
Same, removed the duplicated definition.